modules/lib/core/index.d.ts

```diff
@@ -52,23 +63,71 @@ export interface PublishInfo {
     first?: string;
 }
 
-export interface Component<Config extends object = object, Regions extends Record<string, Region> = Record<string, Region>> {
-    config: Config;
-    descriptor: string;
+export interface Fragment {
+    type: 'fragment'
+    fragment: string;
     path: string;
-    type: 'page' | 'layout' | 'part';
-    regions: Regions;
 }
 
-export interface Region<Config extends object = object> {
+export type LayoutDescriptor = keyof XpLayoutMap;
+export interface Layout<
+    Descriptor extends LayoutDescriptor = LayoutDescriptor
+> {
+    type: 'layout'
+    descriptor: Descriptor
+    config: XpLayoutMap[Descriptor]
+    path?: string // Missing in fragmentPreview https://github.com/enonic/xp/issues/10116
+    regions: Record<string, Region<(Fragment | Part | TextComponent)[]>>;
+}
+
+export type PartDescriptor = keyof XpPartMap;
+export interface Part<
+    Descriptor extends PartDescriptor = PartDescriptor,
+> {
+    type: 'part'
+    descriptor: Descriptor
+    config: XpPartMap[Descriptor]
+    path?: string // Missing in fragmentPreview https://github.com/enonic/xp/issues/10116
+}
+
+export type PageDescriptor = keyof XpPageMap;
+export interface Page<
+    Descriptor extends PageDescriptor = PageDescriptor,
+> {
+    type: 'page'
+    descriptor: Descriptor
+    config: XpPageMap[Descriptor]
+    path: '/'
+    regions: Record<string, Region<(Fragment | Layout | Part | TextComponent)[]>>;
+}
+
+export interface TextComponent {
+    type: 'text'
+    path: string
+    text: string
+}
+
+export interface Component<
+    Config extends object = object,
+    Regions extends Record<string, Region> = Record<string, Region>
+> {
+    config?: Config;
+    descriptor?: string;
+    fragment?: string;
+    path?: string;
+    regions?: Regions;
+    type: 'fragment' | 'layout' | 'page' | 'part' | 'text';
+    text?: string
+}
+
+export interface Region<Components extends Component[] = Component[]> {
     name: string;
-    components: Component<Config>[];
```

Comment:
@tajakobsen I think this is the only "major" change. A region can have multiple different components, so it makes no sense to send in only a single config. To describe the full shape, one needs to pass in all the potential components.

I'm using the very generic Component here, I guess one could use 

```typescript
(Fragment | Layout | Part | TextComponent)[]
```

Instead?

Reply:
I still like the idea of PageRegion and LayoutRegion, since they are not the same.